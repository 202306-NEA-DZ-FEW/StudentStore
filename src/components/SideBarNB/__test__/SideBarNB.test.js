import renderer from "react-test-renderer";

import SidebarNB from "../SideBarNB";

jest.mock("../../../util/firebase", () => {
    return {
        initializeApp: jest.fn(),
    };
});

it("renders correctly", () => {
    const tree = renderer.create(<SidebarNB />).toJSON();
    expect(tree).toMatchSnapshot();
});
