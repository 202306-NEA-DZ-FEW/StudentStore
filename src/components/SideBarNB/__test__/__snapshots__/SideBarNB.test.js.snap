// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="fixed inset-y-0 left-0  w-[60%] sm:w-[40%] md:hidden mx-auto bg-slate-200 shadow-lg transition-transform transform -translate-x-full transition-all ease-in-out duration-300ease-in-out duration-700"
  >
    <div
      className="h-24 shadow-md flex items-center bg-[#C9D9E5]"
    >
      <div
        className="basis-3/4"
      >
        <p
          className="   sm:text-lg  p-2 font-semibold text-[#585785] "
        >
          Hello Samy, Welcome to student store
        </p>
      </div>
    </div>
    <div
      className="p-8 "
    >
      <ul
        className="text-xl space-y-4"
      >
        <li>
          <a
            className="text-[#585785] rounded-md     hover:text-[#FF8A57]  hover:underline hover:decoration-4 font-bold"
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="text-[#585785]  rounded-md  hover:text-[#FF8A57]   hover:underline hover:decoration-4 font-bold"
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Products
          </a>
        </li>
        <li>
          <a
            className="text-[#585785]  hover:text-[#FF8A57] hover:underline hover:decoration-4   rounded-md  font-bold"
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Donation
          </a>
        </li>
        <li>
          <a
            className="text-[#585785]  hover:text-[#FF8A57] rounded-md hover:underline hover:decoration-4  break-keep inline-block font-bold"
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            About us
          </a>
        </li>
      </ul>
      <div
        class="border-b border-gray-400"
      />
      <div
        className="p-2"
      >
        <details
          className=" text-[#585785] font-bold"
        >
          <summary
            className="text-xl"
          >
            Categories
          </summary>
          <ul
            class="text-[#585785]"
          >
            <li>
              <a
                className="capitalize underline hover:text-[#FFA857] transition-all duration-300 ease-in-outtext-lg inline-block ml-4 p-1 md:text-sm md:ml-0 md:py-2 lg:text-md lg:ml-0  "
                href="#"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                All Categories
              </a>
            </li>
            <li>
              <a
                className="capitalize hover:text-[#FFA857] transition-all duration-300 ease-in-out  text-lg inline-block ml-4 p-1 md:text-sm md:ml-0 md:py-2 lg:text-md lg:ml-0  "
                href="#"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Electronics
              </a>
            </li>
            <li>
              <a
                className="capitalize hover:text-[#FFA857] transition-all duration-300 ease-in-out text-lg inline-block ml-4 p-1 md:text-sm md:ml-0 md:py-2 lg:text-md lg:ml-0  "
                href="#"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Books
              </a>
            </li>
            <li>
              <a
                className="capitalize hover:text-[#FFA857] transition-all duration-300 ease-in-out  text-lg inline-block ml-4 p-1 md:text-sm md:ml-0 md:py-2 lg:text-md lg:ml-0  "
                href="#"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Gaming
              </a>
            </li>
            <li>
              <a
                className="capitalize hover:text-[#FFA857] transition-all duration-300 ease-in-out  text-lg inline-block ml-4 p-1 md:text-sm md:ml-0 md:py-2 lg:text-md lg:ml-0  "
                href="#"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Clothes
              </a>
            </li>
            <li>
              <a
                className="capitalize hover:text-[#FFA857] transition-all duration-300 ease-in-out  text-lg inline-block ml-4 p-1 md:text-sm md:ml-0 md:py-2 lg:text-md lg:ml-0  "
                href="#"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Shoes
              </a>
            </li>
            <li>
              <a
                className="capitalize hover:text-[#FFA857] transition-all duration-300 ease-in-out  text-lg inline-block ml-4 p-1 md:text-sm md:ml-0 md:py-2 lg:text-md lg:ml-0  "
                href="#"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Food
              </a>
            </li>
            <li>
              <a
                className="capitalize hover:text-[#FFA857] transition-all duration-300 ease-in-out  text-lg inline-block ml-4 p-1 md:text-sm md:ml-0 md:py-2 lg:text-md lg:ml-0  "
                href="#"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Transportation
              </a>
            </li>
            <li>
              <a
                className="capitalize hover:text-[#FFA857] transition-all duration-300 ease-in-out  text-lg inline-block ml-4 p-1 md:text-sm md:ml-0 md:py-2 lg:text-md lg:ml-0  "
                href="#"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Furniture
              </a>
            </li>
          </ul>
        </details>
      </div>
    </div>
    <button
      className="absolute top-4 cursor-pointer right-4 text-[#585785] hover:text-[#FFA857]  text-5xl"
    >
      Ã—
    </button>
  </div>
  <div
    className=""
  >
     
  </div>
</div>
`;
